components/script/dom/htmlinputelement.rs

```diff
@@ -410,8 +410,19 @@ impl HTMLInputElementMethods for HTMLInputElement {
     fn SetValue(&self, value: DOMString) -> ErrorResult {
         match self.value_mode() {
             ValueMode::Value => {
+                // Step 1.
+                let old_value = self.textinput.borrow().get_content();
```

Comment:
Merge these 2 steps together so that the older value doesn't get cloned.

Reply:
We need to clone the old value so that we may compare it in step 5.

Reply:
No you don't, merge these two steps and use `mem::replace` and you don't need the clone.